src/Modules/OrchardCore.Commerce.Payment.Stripe/Migrations/StripeMigrations.cs

```diff
@@ -0,0 +1,39 @@
+using Lombiq.HelpfulLibraries.OrchardCore.Data;
+using OrchardCore.Commerce.Payment.Stripe.Indexes;
+using OrchardCore.Commerce.Payment.Stripe.Models;
+using OrchardCore.ContentManagement.Metadata;
+using OrchardCore.ContentManagement.Metadata.Settings;
+using OrchardCore.Data.Migration;
+using YesSql.Sql;
+using static OrchardCore.Commerce.Abstractions.Constants.ContentTypes;
+
+namespace OrchardCore.Commerce.Payment.Stripe.Migrations;
+
+public class StripeMigrations : DataMigration
+{
+    private readonly IContentDefinitionManager _contentDefinitionManager;
+
+    public StripeMigrations(IContentDefinitionManager contentDefinitionManager) =>
+        _contentDefinitionManager = contentDefinitionManager;
+
+    public int Create()
+    {
+        _contentDefinitionManager
+            .AlterPartDefinition<StripePaymentPart>(builder => builder
+                .Configure(part => part.Attachable())
+                .WithField(part => part.PaymentIntentId));
+
+        _contentDefinitionManager
+            .AlterTypeDefinition(Order, builder => builder
+                .WithPart(nameof(StripePaymentPart)));
+
+        // This table may exist when migrating from an old version of the DB where it was created in a different module.
+        SchemaBuilder.DropMapIndexTable(typeof(OrderPaymentIndex));
```

Comment:
I don't think this can cause any problems, because this index is really only used as a temporary non-volatile storage during the payment process. We don't need a historic record of this.